groovy-language-server/src/test/java/com/palantir/ls/groovy/GroovyLanguageServerIntegrationTest.java

```diff
@@ -0,0 +1,419 @@
+        InitializeResult result = completableResult.get(60, TimeUnit.SECONDS);
+        assertCorrectInitializeResult(result);
+
+        // Send a didOpen request to trigger a compilation
+        sendDidOpen(test1);
+
+        // Give it some time to publish
+        Thread.sleep(2000);
+
+        Set<MyPublishDiagnosticParams> expectedDiagnosticsResult =
+                Sets.newHashSet(
+                        new MyPublishDiagnosticParams(test1.toPath().toUri().toString(),
+                                Sets.newHashSet(new DefaultDiagnosticBuilder(
+                                        "unable to resolve class ExceptionNew1 \n @ line 7, column 18.",
+                                        DiagnosticSeverity.Error)
+                                                .range(Ranges.createRange(6, 17, 6, 72))
+                                                .source(test1.getAbsolutePath())
+                                                .build())),
+                        new MyPublishDiagnosticParams(test2.toPath().toUri().toString(),
+                                Sets.newHashSet(new DefaultDiagnosticBuilder(
+                                        "unable to resolve class ExceptionNew222 \n @ line 7, column 18.",
+                                        DiagnosticSeverity.Error)
+                                                .range(Ranges.createRange(6, 17, 6, 74))
+                                                .source(test2.getAbsolutePath())
+                                                .build())));
+        assertEquals(expectedDiagnosticsResult, publishedDiagnostics.stream().collect(Collectors.toSet()));
+        assertEquals(2, publishedDiagnostics.size());
+    }
+
+    private void sendDidOpen(File file) {
+        client.getTextDocumentService()
+                .didOpen(new DidOpenTextDocumentParamsBuilder()
+                        .textDocument(
+                                new TextDocumentItemBuilder()
+                                        .languageId("groovy")
+                                        .uri(file.getAbsolutePath())
+                                        .version(0)
+                                        .text("foo")
+                                        .build())
+                        .build());
+    }
+
+    private void assertCorrectInitializeResult(InitializeResult result) {
+        ServerCapabilities expectedCapabilities = new ServerCapabilitiesBuilder()
+                .textDocumentSync(TextDocumentSyncKind.Incremental)
+                .documentSymbolProvider(true)
+                .workspaceSymbolProvider(true)
+                .referencesProvider(true)
+                .completionProvider(new CompletionOptionsBuilder()
+                        .resolveProvider(false)
+                        .triggerCharacter(".")
+                        .build())
+                .build();
+        assertEquals(Sets.newHashSet(
```

Comment:
move this to the end of this function to keep all the lines related to serverCapabilities together